### PR TITLE
Added sharing via Facebook

### PR DESCRIPTION
**Pull request by clintonb**
Includes #193

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/ecommerce please review

File: credentials/apps/core/admin.py [hunk 1]
```diff
@@ -23,6 +26,29 @@ class SiteConfigurationAdmin(admin.ModelAdmin):
...
+        }),
+        (_('Social Sharing'), {
+            'fields': (
+                'facebook_app_id', 'twitter_username', 'enable_facebook_sharing', 'enable_linkedin_sharing',
+                'enable_twitter_sharing',
+            )
+        }),
+    )
+
+    def save_model(self, request, obj, form, change):
```
Comment (clintonb) [inline credentials/apps/core/admin.py]: FYI @mjfrey @douglashall @mattdrayer this should be incorporated into ecommerce and other services where site caching is an issue.

Comment (douglashall) [inline credentials/apps/core/admin.py]: @clintonb This works great in an environment running a single gunicorn work process. In a multiprocess environment, not so much. Consider installing the latest version of edx-django-sites-extensions which provides a more robust solution to this problem (Set a TTL on the Site cache). Note that ecommerce currently uses an older version of edx-django-sites-extensions which does not include the TTL on the Site cache. We should updgrade to the latest version of edx-django-sites-extensions so we can take advantage of the TTL.

Comment (clintonb) [inline credentials/apps/core/admin.py]: Using `edx_django_sites_extensions` is a non-starter due to https://github.com/edx/edx-django-sites-extensions/issues/11.

Comment (douglashall) [inline credentials/apps/core/admin.py]: Hmm, I think https://github.com/edx/edx-django-sites-extensions/blob/master/README.rst contains the information you are looking for. Let me know if I can help clarify anything.

File: credentials/settings/base.py [hunk 2]
```diff
@@ -43,6 +42,7 @@
     'social.apps.django_app.default',
     'waffle',
     'storages',
+    'webpack_loader',
```
Comment (mikedikan) [inline credentials/settings/base.py]: why was this moved to 3rd party?

File: docs/credentials_admin.rst [hunk 3]
```diff
@@ -25,92 +25,21 @@ Each site can have only one configuration.
 
 
 Add Another Site and Site Configuration
------------------------------------------
-
-If you want to add more sites, partners, and site configurations, you can use
-the ``create_or_update_site`` command. The following options are available for
-this command.
```
Comment (mikedikan) [inline docs/credentials_admin.rst]: why was this all removed?

File: credentials/settings/base.py (hunk 2 above)
Comment (clintonb) [inline credentials/settings/base.py]: It's a third-party app.

File: docs/credentials_admin.rst (hunk 3 above)
Comment (clintonb) [inline docs/credentials_admin.rst]: It's not needed, and will grow stale over time. If you need to know the command args, use `--help`.

File: credentials/settings/base.py (hunk 2 above)
Comment (mikedikan) [inline credentials/settings/base.py]: and it didn't used to be?

Comment (clintonb) [inline credentials/settings/base.py]: It always has been. It was put in the wrong place initially.
